src/main/java/org/kohsuke/github/GHIssue.java

```diff
@@ -360,17 +360,11 @@ public void addLabels(Collection<GHLabel> labels) throws IOException {
     }
 
     private void _addLabels(Collection<String> names) throws IOException {
-        List<String> newLabels = new ArrayList<String>();
-
-        for (GHLabel label : getLabels()) {
-            newLabels.add(label.getName());
-        }
-        for (String name : names) {
-            if (!newLabels.contains(name)) {
-                newLabels.add(name);
-            }
-        }
-        setLabels(newLabels.toArray(new String[0]));
+        root.createRequest()
+                .with("labels", names.toArray(new String[0]))
```

Comment:
Wait... this should add a list of labels in one call, not only work for one label. 
https://docs.github.com/en/rest/reference/issues#add-labels-to-an-issue

Reply:
I think that's what it did anyway but I see you can pass directly the collection so that's what is done now.